chrome/content/zotero/xpcom/sync/syncAPIClient.js

```diff
@@ -430,6 +429,54 @@ Zotero.Sync.APIClient.prototype = {
 	}),
 	
 	
+	getAPIKeyFromCredentials: Zotero.Promise.coroutine(function* (username, password) {
+		var apiKey;
+		var body = JSON.stringify({
+			username: username,
+			password: password,
+			name: "Automatic Zotero Client Key",
+			access: {
+				user: {
+					library: true,
+					notes: true,
+					write: true,
+					files: true
+				},
+				groups: {
+					all: {
+						library: true,
+						write: true
+					}
+				}
+			}
+		});
+		var headers = {
+			"Content-Type": "application/json"
+		};
+		var uri = this.baseURL + "keys";
+		var response = yield this.makeRequest("POST", uri, {
+			body, headers, successCodes: [201, 403], noAPIKey: true
+		});
+		if (response.status == 403) {
+			return false;
+		}
+		
+		var json = this._parseJSON(response.responseText);
+		apiKey = json.key;
+		if (!apiKey) {
+			return false;
+		}
```

Comment:
If there's no `key` property, it should just throw, since that should never happen.